### ISSUE TITLE
replace MinMaxPriorityQueue (and remove guava dependency)?

### ISSUE THREAD

**Issue by davidmoten** [davidmoten/rtree#39]
I read this stackoverflow [report](http://stackoverflow.com/questions/7878026/is-there-a-priorityqueue-implementation-with-fixed-capacity-and-custom-comparato) with comments from google employee (apparently) that other implementations would be significantly more performant than `MinMaxPriorityQueue`:

>the constant factors will be significantly worse than a normal PriorityQueue. Using a normal PriorityQueue will do much better, or better yet, Ordering.greatestOf uses an O(n) time, O(k) memory algorithm. (We are giving some consideration to deprecating MinMaxPriorityQueue, just because it tends to get misused in this way.) – Louis Wasserman Aug 21 '15 at 17:33 

I'll set up a JMH benchmark and experiment with other implementations. If anyone has one that they like I'd love to know, thanks!

**Comment by davidmoten** [comment]
commit 46612826f07b2af22b00276c70ec7301fe85d5c6 contains removal of all dependencies on guava except for the `MinMaxPriorityQueue` used by `RTree.nearest`.

**Comment by davidmoten** [comment]
`PriorityQueue(comparator)` not available pre Java 8 so may have to backport into this project.

**Comment by davidmoten** [comment]
Java 8 backport of `PriorityQueue` used inside a custom `BoundedPriorityQueue` class gives 1.7x throughput of Guava `MinMaxPriorityQueue`:
```
Benchmark                            Mode  Cnt     Score    Error  Units
Guava MaxMinPriorityQueue:
BenchmarksRTree.searchNearestGreek  thrpt   30  2410.815 ± 56.740  ops/s
Java 8 backport of PriorityQueue + BoundedPriorityQueue:
BenchmarksRTree.searchNearestGreek  thrpt   30  4091.053 ± 203.721  ops/s
```

**Comment by davidmoten** [comment]
Released in 0.7.5 (which has been re-released as 0.7.6 with a couple of bug fixes)
